core/src/main/java/org/apache/iceberg/avro/AssignFieldIds.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.avro;
+
+import com.google.common.base.Preconditions;
+import java.util.List;
+import java.util.Map;
+import org.apache.avro.Schema;
+
+/**
+ * Assign field ids to the Avro schema using the given field name to field id map
+ */
+public class AssignFieldIds extends AvroSchemaVisitor<Void> {
```

Comment:
Shouldn't this return `Schema`?

Reply:
I'm updating the fieldIds in place in the passed in Schema. Avro allows it. So the 'fileSchema' is being modified instead of creating a new one.

Reply:
I think this should create a new one instead of modifying what was passed in.

Reply:
OK, I'll update